### PR TITLE
Range summary

### PR DESCRIPTION
**Pull request by christophsax**

### CODE REVIEW

**Comment by christophsax** [comment]
@benubah Somehting like that? Does it make sense to you?

**Comment by benubah** [comment]
Thanks! Looks good. But it breaks for me:

```
Warning: Error in : Can't subset columns that don't exist.
x Column `group` doesn't exist.
  122: <Anonymous>
```

Checking where its from

**Comment by benubah** [comment]
@christophsax The error is **not** from your solution. Rather we no longer needed this redundant line (https://github.com/dsbbfinddx/shinyfindapps/blob/main/map/R/server.R#L125) in `server.R` in response to the changes.

File: R/summraize_over_group_and_time_range.R [hunk 1]
```diff
@@ -13,58 +13,80 @@ summarize_over_group_and_time_range <- function (data_all, country_last_update_i
...
-      sum_all_new_tests      = sum(all_new_tests              , na.rm = TRUE),
-      avg_pos                = avg_all_new_cases / avg_all_new_tests,
-      avg_cap100k_new_cases  = mean(all_new_cases / pop_100k  , na.rm = TRUE),
-      sum_cap100k_new_cases  = sum(all_new_cases / pop_100k   , na.rm = TRUE),
-      avg_cap100k_new_deaths = mean(all_new_deaths / pop_100k , na.rm = TRUE),
-      sum_cap100k_new_deaths = sum(all_new_deaths / pop_100k  , na.rm = TRUE),
-      avg_cap100k_new_tests  = mean(all_new_tests / pop_100k  , na.rm = TRUE),
-      sum_cap100k_new_tests  = sum(all_new_tests / pop_100k   , na.rm = TRUE),
+      # group = group[1],
+      pop_100k                 = mean(pop_100k, na.rm = TRUE),
```
Comment (dsbb-finddx) [inline R/summraize_over_group_and_time_range.R]: Why do we need to have a mean for the population?
The population of each country is unique, so it should be used in the group_by

File: R/summraize_over_group_and_time_range.R [hunk 2]
```diff
@@ -13,58 +13,80 @@ summarize_over_group_and_time_range <- function (data_all, country_last_update_i
...
+      pop                      = mean(pop, na.rm = TRUE),
```
Comment (dsbb-finddx) [inline R/summraize_over_group_and_time_range.R]: Same here as above

File: R/summarize_over_group_and_time_range.R [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      as.Date(end_date)
+    ))
+# browser()
+  data_summarized <-
+    data_filtered |>
+    group_by(unit) |>
+    summarize(
+      name = name[1],
+      # group = group[1],
+      pop_100k                 = mean(pop_100k, na.rm = TRUE),
```
Comment (dsbb-finddx) [inline R/summarize_over_group_and_time_range.R:29]: @christophsax @benubah why do we neeb the mean of the population?
Can we replace either by adding pop100K and pop in the group_by?
Or by adding: 
```{r} 
pop_100K = pop_100K[1],
pop = pop[1]

```

Comment (christophsax) [reply R/summarize_over_group_and_time_range.R:29]: It's a constant, so this should be equivalent. Will try to put all aggregations in a single function with a reasonable documentation. Let's review this afterward.

**Pull request merged**